geopandas/geodataframe.py

```diff
@@ -755,16 +765,43 @@ def concat(L, axis=0, ignore_index=False):
     geometry = vectorized.concat(df._geometry_array for df in L)
     if isinstance(L[0], GeoDataFrame):
         L = [df.drop(name, axis=1) for df in L]
-        new = pd.concat(L, ignore_index=ignore_index)
+        new = pd.concat(L, **kwargs)
         new = new.set_geometry(GeoSeries(geometry, name=name, index=new.index),
                                crs=L[0].crs)
     else:
         index = pd.concat([pd.Series(index=s.index) for s in L],
-                          ignore_index=ignore_index).index
+                          **kwargs).index
         new = GeoSeries(geometry, index=index, name=name, crs=L[0].crs)
     return new
 
 
+def _concat_axis1(objs, **kwargs):
+    is_geo = [isinstance(obj, (GeoDataFrame, GeoSeries)) for obj in objs]
+
+    if not sum(is_geo) == 1:
+        raise ValueError("'concat' with axis=1 currently only supports "
+                         "concatenating objects of which only one is a "
+                         "GeoDataFrame or GeoSeries")
+
+    geo_obj = objs[is_geo.index(True)]
+    if isinstance(geo_obj, GeoSeries):
+        geometry = geo_obj
+        temp_obj = pd.Series(0, index=geo_obj.index, name=geo_obj.name)
+    else:
+        geometry = geo_obj.geometry
+        temp_obj = geo_obj.copy()
+        temp_obj[geometry.name] = 1
```

Comment:
Why set this to a dummy value rather than dropping the column?

Reply:
If I recall correctly, it was to keep the order of the columns intact.